CONTRIBUTING.md

```diff
@@ -13,53 +13,114 @@ At the same time, I had a day job where I was a junior software developer at a r
 
 # Contributor Manual
 
 ## Prerequisite
 
-```shell
-node: "^12.20.0 || ^14.13.1 || >=16.0.0"
-yarn: "^1.22.10"
-# otherwise, your build will fail
-```
+You’ll need the following tools set up on your system to contribute to this codebase:
 
-## Setting up your local repo
+- [Node 14.x or 16.x][install-node]
+- [Yarn 1.x][install-yarn] (_Note: we do NOT use Yarn 2!_)
+- [Git 2.x][install-git]
 
-Astro uses yarn workspaces, so you should **always run `yarn install` from the top-level project directory.** running `yarn install` in the top-level project root will install dependencies for `astro`, `www`, `docs`, and every package in the repo.
+## Setting up Astro locally
 
-```shell
-git clone && cd ...
-yarn install
-yarn build:all
+All work on Astro begins by downloading a copy locally. Start by [**forking**][how-to-fork] this repository (if you’re new to this, [GitHub has a great detailed guide][how-to-fork]). You should end up with an `astro/` code folder on your local machine.
+
+Switch to this folder by opening your **Terminal application** of choice and running:
+
+```
+cd /[path to my code folder]/astro
 ```
 
+Next, you’ll need to install dependencies and set up the project. You can do that by running the following commands from within the `astro/` folder:
+
+```
+yarn
+yarn build
+```
+
+This will download everything you need and build Astro locally for use. You’re ready to go!
+
 ## Development
 
-```shell
-# starts a file-watching, live-reloading dev script for active development
+While working within packages (such as `packages/astro/*`), you may notice you have to run `yarn build` to see your changes. This process can really slow you down if you are trying many changes at once. Try running the following instead:
+
+```
 yarn dev
-# build the entire project, one time.
+```
+
+This will start a build watcher, which will rebuild files as you work. Use this to speed up your workflow, and see your changes more rapidly.
+
+Of course, you can still build all files within a package, as before. You may need to do so from time to time:
+
+```
 yarn build
 ```
 
 ## Running tests
 
-```shell
-# run this in the top-level project root to run all tests
+You’ll need to run existing tests to make sure your work didn’t cause any regressions. To create a fresh build and run all tests, run the following from the project root:
+
+```
+yarn
+yarn build
 yarn test
-# run only a few tests, great for working on a single feature
-# (example - `yarn test rss` runs `astro-rss.test.js` tests)
-yarn test $STRING_MATCH
 ```
 
+If you find yourself wanting to not run the whole test suite, you can run the following instead:
+
+```
+cd packages/astro
+yarn test astro-markdown
+```
+
+Replace “`astro-markdown`“ with any part of a test filename to only run those tests.
+
+## Making a Pull Request
```

Comment:
some questions:

1. Most developers can't push directly to `origin`, so this needs to be updated to describe how Git/Github forks work.
2. Asking more broadly: Can we link to some good GitHub docs on this topic instead? I feel like this is a bit out of scope for our contributor docs.
3. Asking even more broadly: Are we able to take on contributions from people who aren't familiar with how GitHub works? I think the answer is yes, but I just want to be sure.

my 2 cents: I actually like that a developer doesn't currently need to worry about changesets until _after_ they have already submitted the PR and get the warning from the bot. It reduces the time to "I did it! success!" The changeset work becomes its own small followup task, with no impact on the already steep learning curve. Bonus: the bot documents how to do it, maintains that documentation, etc. so that we don't have to.

The other comments are more nits and personal preference, but this one I do feel a bit more strongly about.

Reply:
Didn’t the document above link to a guide on forking and start from the standpoint of them working from their own fork?

Reply:
oh right, `origin` in this case is their fork and not ours. nvm then on num 1! The rest of the comments/questions still stand.